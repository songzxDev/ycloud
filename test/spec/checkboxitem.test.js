import ko from 'knockout'
import comp from '../../src/components/checkbox/checkboxitem.js'
import $ from 'jquery'
describe('y-checkboxitem组件', () => {
  const PREFIX = 'y-'
  ko.components.register(PREFIX + comp.name, {
    viewModel: comp.init,
    template: comp.template,
    synchronous: true
  })
  ko.cleanNode(document.getElementsByTagName('body')[0])
  let vm = {
    title: ko.observable('test'),
    item: {value: 1,label:'男'},
    valueList: ko.observableArray([1])
  }
  document.body.innerHTML = '<y-checkbox-item params="label: item.label, value: item.value, valueList: valueList"></y-checkbox-item>'
  ko.applyBindings(vm)
  var el = $('body')
  it("组件注册测试", () => {
    expect(comp).not.toBeNull();
    expect(ko.components.isRegistered(PREFIX + comp.name)).toBe(true);
  })
  it('生成checkboxitem测试', () => {
    expect($('body').find('.y-checkbox-item').length > 0).toEqual(true)
  })
  it('默认值设置测试', () => {
    expect($('body').find('.y-checkbox-item').hasClass('y-checkbox-item-checked')).toEqual(true)
  })
  // todo radio方向测试
  it('快照', () => {
    expect($('body').get(0)).toMatchSnapshot()
  })
})
