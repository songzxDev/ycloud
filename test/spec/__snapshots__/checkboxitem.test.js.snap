// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`y-checkboxitem组件 快照 1`] = `
<body>
  <y-checkbox-item
    params="label: item.label, value: item.value, valueList: valueList"
  >
    <label
      class="y-checkbox-item y-checkbox-item-checked"
      data-bind="css:{'y-checkbox-item-checked': checked}"
    >
      
  
      <span
        class="y-checkbox-item--inner-warpper"
      >
        
    
        <span
          class="y-checkbox-item-inner"
        >
          
      
          <input
            data-bind="event:{click:hanldeClick}"
            type="checkbox"
          />
          
    
        </span>
        
  
      </span>
      
  
      <span
        class="y-checkbox-label"
        data-bind="text:label"
      >
        男
      </span>
      

    </label>
    

  </y-checkbox-item>
</body>
`;
